lib/autoPipelining.ts

```diff
@@ -68,8 +72,13 @@ function executeAutoPipeline(client, slotKey: string) {
   });
 }
 
-export function shouldUseAutoPipelining(client, commandName: string): boolean {
+export function shouldUseAutoPipelining(
+  client,
+  functionName: string,
+  commandName: string
+): boolean {
   return (
+    functionName &&
```

Comment:
In general looks fine to me.
The only doubt I have is here: why you also check for functionName here?

Reply:
commands created by `createBuiltinCommand` would have no functionName, this would prevent the auto pipelining breaking for commands created in this way

Reply:
I see.
But why do you want to prevent autopipeline to run against them? Autopipeline could support them.

Reply:
I don't want to prevent, but the best way i can see to support it without creating an API that exposes the internals of this is via the `addBultinCommand` method i have created

Reply:
👍